octodns/provider/dyn.py

```diff
@@ -383,6 +396,10 @@ def populate(self, zone, target=False, lenient=False):
         self.log.info('populate:   found %s records',
                       len(zone.records) - before)
 
+        if self.cleanup_session:
+            _CachingDynZone.flush_zone(zone.name[:-1])
```

Comment:
There's 2 more "cache" vars on the provider object around traffic directors. They'd likely need to be cleared too if we go this route.

https://github.com/markrechler/octodns/blob/c45b7a5fb0c4e75987a866667de242ad78858fe1/octodns/provider/dyn.py#L154-L156

Those should probably go into a `_reset_caches` function or something like which is called from the bottom of `__init__` as well as any other point where things need to be reset.